alpaca/broker/client.py

```diff
@@ -524,10 +524,9 @@ def _get_account_activities_iterator(
             # ok we made it to the end, we need to ask for the next page of results
             last_result = result[-1]
 
-            if "id" not in last_result:
-                raise APIError(
-                    "AccountActivity didn't contain an `id` field to use for paginating results"
-                )
+            assert (
+                "id" in last_result
+            ), "AccountActivity didn't contain an `id` field to use for paginating results"
```

Comment:
`assert` is ignored if optimization is requested. Therefore, if we need to handle an error, we should not use `assert` as a error handling method.

ref. https://docs.python.org/3.12/reference/simple_stmts.html#the-assert-statement